spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala

```diff
@@ -91,7 +91,16 @@ case class DeltaTableV2(
   // in cases where we will fallback to the V1 behavior.
   lazy val deltaLog: DeltaLog = {
     DeltaTableV2.withEnrichedUnsupportedTableException(catalogTable, tableIdentifier) {
-      DeltaLog.forTable(spark, rootPath, options)
+      // Ideally the table storage properties should always be the same as the options load from
+      // the Delta log, as Delta CREATE TABLE command guarantees it. However, custom catalogs such
+      // as Unity Catalog may add more table storage properties on the fly. We should respect it
+      // and merge the table storage properties and Delta options.
+      val dataSourceOptions = if (catalogTable.isDefined) {
+        DataSourceUtils.generateDatasourceOptions(caseInsensitiveOptions, catalogTable.get)
```

Comment:
From where is `options` variable passed to DeltaTableV2? Is that from: `spark.read.options(k1, v1).load(<path>)` ?

If the above is true, should we be respecting options even in case when `catalogTable` is defined?

i.e. 

```
        options ++ DataSourceUtils.generateDatasourceOptions(caseInsensitiveOptions, catalogTable.get)
```

Reply:
This is `DeltaTableV2`, so the `options` was loaded from the delta log.

Reply:
No - DeltaTableV2 options don't come from DeltaLog. They can come from `.options()` API as well. This change introduced these options: https://github.com/delta-io/delta/commit/4afd4156680ef37c9e8509cebab10ad43a0755f4

Reply:
The case here is `catalogTable` is present. This means the table exists and the `options` should be loaded from the Delta log?

Reply:
> The case here is catalogTable is present.

Yes I understand that. DeltaLog directory does not store any options.

A user can pass options explcitly as part of command. e.g.

    spark.read.option(k1, v1).load("\<path\>") -- definitely works and options are tunneled to DeltaLog object properly

    spark.read.option(k1, v1).table("c1.s1.t1") -- most like options are tunnel properly today

Reply:
> should we be respecting `options` even in case when `catalogTable` is defined?

Isn't `caseInsensitiveOptions` just a different representation of `options`?
```scala
private lazy val caseInsensitiveOptions = new CaseInsensitiveStringMap(options.asJava)  
```
(so I think it's included already by the current code)

Meanwhile tho:
> `spark.read.option(k1, v1).table("c1.s1.t1")` -- most like `options` are tunnel properly today

IIRC, options passed by `option` are _not_ propagated through `table`. At least, not all of them. It prevents option-based time travel on a catalog table, for example.

Reply:
> Isn't caseInsensitiveOptions just a different representation of options?

Oh right. I missed this part. So we are already respective passed `options`.

Reply:
is this thread resolved?